File: solutions/0053.maximum-subarray.ts
// https://leetcode.com/problems/maximum-subarray
function maxSubArray(nums: number[]): number {
  const numsLength = nums.length;
  let maxSum = nums[0];
  let tmpSum = nums[0];

  for (let i = 1; i < numsLength; i++) {
    tmpSum = Math.max(tmpSum + nums[i], nums[i]);

    if (maxSum < tmpSum) {
      maxSum = tmpSum;
    }
  }

  return maxSum;
}

File: solutions/0053.maximum-subarray.go
// https://leetcode.com/problems/maximum-subarray
func maxSubArray(nums []int) int {
	var numsLength int = len(nums)
	var maxSum int = -10000

	for i := 0; i < numsLength; i++ {
		var tmpSum int = 0

		for j := i; j < numsLength; j++ {
			tmpSum += nums[j]

			if maxSum < tmpSum {
				maxSum = tmpSum
			}
		}
	}

	return maxSum
}
